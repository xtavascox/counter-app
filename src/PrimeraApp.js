import React from "react";
import PropTypes from 'prop-types'

export const PrimeraApp = ({saludo,subtitulo}) => {
 
  return (
    <>
    
    <h1>{saludo}!!!!</h1>
      {/* <pre>{JSON.stringify(datos,null,3)}</pre> */}
      <p>{subtitulo}</p>


      
    </>
  );
};

PrimeraApp.propTypes={
    saludo:PropTypes.string.isRequired
}
PrimeraApp.defaultProps={
    subtitulo:'soy un subtitulo default'
}
export default PrimeraApp;
