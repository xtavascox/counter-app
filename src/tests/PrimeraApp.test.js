import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import React from "react";
import PrimeraApp from "../PrimeraApp";

describe("Pruebas en <PrimeraAp />", () => {
  
    test("debe de mostrar <PrimeraApp /> correctamente", () => {
    const saludo = "hola,soy goku";
    const wrapper = shallow(<PrimeraApp saludo={saludo} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('debe de mostrar el subtitulo enviado por props',()=>{
    const saludo = "hola,soy goku";
    const subtitulo='puto el que lo lea'
    const wrapper = shallow(<PrimeraApp saludo={saludo} subtitulo={subtitulo} /> )
    const textoParrafo=wrapper.find('p').text()
    expect(textoParrafo).toBe(subtitulo)
    
  })
});
