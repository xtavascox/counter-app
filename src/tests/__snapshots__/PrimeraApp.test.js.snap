// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraAp /> debe de mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    hola,soy goku
    !!!!
  </h1>
  <p>
    soy un subtitulo default
  </p>
</Fragment>
`;
